### ISSUE TITLE
Z2M constantly offers an OTA update while the module is up to date.

### ISSUE THREAD

**Issue by tadikwa** [Koenkk/zigbee2mqtt#20722]
### What happened?

It was noted that Z2M does not report the correct value when it comes to the firmware version.
Indeed, the latter constantly indicates that the module will be in V1 while it is in V13.
Therefore, home assistant constantly offers the OTA update.

In the "About" tab it correctly displays the firmware value 4000-0013.
On the other hand, in "Status", it indicates the firmware in version 1.

It seems to me that the latter lists the "hwVersion" field instead of the "swBuildId" field

### What did you expect to happen?

The expected behavior is that Z2M considers the Lixee module up to date

### How to reproduce it (minimal and precise)

To reproduce the phenomenon, simply associate the Lixee Zlinky module with a zigate platform so that it is available in Z2M and note in the OTA tab that an update is available, and also check the tab "Status" which states the wrong firmware version.

### Zigbee2MQTT version

1.34.0

### Adapter firmware version

a003

### Adapter

LiXee ZLinky_TIC

### Setup

HASS.io on ESX Server

### Debug log

Read result of 'genBasic': {"hwVersion":1,"swBuildId":"4000-0013"}

**Comment by tadikwa** [comment]
![2024-01-10_17h02_34](https://github.com/Koenkk/zigbee2mqtt/assets/6054289/c4f50f0f-66c8-4cd8-a209-4b08f0ba51df)
![2024-01-10_17h02_43](https://github.com/Koenkk/zigbee2mqtt/assets/6054289/62d3f74c-9323-47a2-877f-3d36400bd869)
![2024-01-10_17h02_52](https://github.com/Koenkk/zigbee2mqtt/assets/6054289/9799617f-98fc-4c6d-ac61-02f9711ecc3d)

**Comment by MilesTEG1** [comment]
Hello,
I've got a similar problem, except I have a limited firmware version (ZLinky_router_v13_limited.bin), I flashed this WE over the legacy version.
The ZLinly swBuildId is `4001-0013` whereas the hwVersion is `1`.
![CleanShot 2024-03-18 at 09 19 39@2x](https://github.com/Koenkk/zigbee2mqtt/assets/29298312/f0825a93-ef6a-48f6-9acc-d59cb406a549)

And Z2M see an update that can't be done via OTA.

ANd what's strange is that the "About" page doesn't show the right version of the firmware :
![CleanShot 2024-03-18 at 09 20 14@2x](https://github.com/Koenkk/zigbee2mqtt/assets/29298312/ceed17f3-b7b1-4a3c-9167-63885e336aff)
It should be seen as `4001-0013` not `4000-0013`.

Can this be fixed?
And OTA update disable if we have an OTA firmware while OTA for limited firmware isn't supported ?
